src/field_theory/krull_topology.lean

```diff
@@ -287,3 +289,363 @@ begin
+  `E →ₐ[K] L`. This Lemma tells us that the underlying maps of these two terms agree on `E`.
+    -/
+lemma equiv_restricts_to_alg_hom_of_finite_dimensional (h_int : algebra.is_integral K L)
+(f : ultrafilter (L →ₐ[K] L)) {E : intermediate_field K L} (h_findim : finite_dimensional K E) :
+((ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int f).to_alg_hom.comp E.val) =
+f.generator_of_pushforward h_findim :=
+begin
+  ext,
+  have h : ((ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int f).to_alg_hom.comp E.val)
+  x =
+  (ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int f).to_fun x := rfl,
+  rw h,
+  exact ultrafilter.glued_generators_of_pushforwards_function_spec' _ _ _ _,
+end
+
+lemma inv_mul_alg_equiv_of_elem (x : L) (f g : L ≃ₐ[K] L) :
+(f⁻¹ * g) x = x ↔ g x = f x :=
+begin
+   rw alg_equiv.mul_apply,
+   split,
+   { intro h,
+    have h' := congr_arg f h,
+    rw [← alg_equiv.mul_apply, mul_right_inv] at h',
+    exact h' },
+   { intro h,
+    have h' := congr_arg f.symm h,
+    rw alg_equiv.symm_apply_apply at h',
+    exact h' },
+end
+
+
+lemma top_group_map_nhds_eq {G : Type*} [group G] [topological_space G]
+[topological_group G] (g x : G) :
+filter.map (λ y, g * y) (nhds x) = nhds (g * x) :=
+map_mul_left_nhds g x
+
+
+lemma ultrafilter_converges_to_glued_equiv (h_int : algebra.is_integral K L)
+  (f : ultrafilter (L ≃ₐ[K] L)) :
+(f : filter (L ≃ₐ[K] L)) ≤ nhds (ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int
+(f.map (λ (σ : L ≃ₐ[K] L),
+σ.to_alg_hom))) :=
+begin
+  let σ := ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int (f.map (λ (e : L ≃ₐ[K] L),
+  e.to_alg_hom)),
+  intros A hA,
+  have hA_coset : left_coset σ⁻¹ A ∈ nhds (1 : L ≃ₐ[K] L),
+  { have h_sigma_1 : σ = σ * 1 := by simp,
+    change A ∈ nhds σ at hA,
+    rw [h_sigma_1, ← top_group_map_nhds_eq σ 1, filter.mem_map] at hA,
+    have h : left_coset σ⁻¹ A = (λ y, σ * y)⁻¹' A,
+    { ext,
+      split,
+      { rintro ⟨a, ha, hax⟩,
+        simp [hax.symm, ha] },
+      { intro hx,
+        rw set.mem_preimage at hx,
+        rw [mem_left_coset_iff, inv_inv],
+        exact hx } },
+    rw h,
+    exact hA },
+  have hA_coset_cont_H : ∃ (E : intermediate_field K L), finite_dimensional K E
+  ∧ E.fixing_subgroup.carrier ⊆ left_coset σ⁻¹ A,
+  { rw [group_filter_basis.nhds_one_eq, filter_basis.mem_filter_iff] at hA_coset,
+    rcases hA_coset with ⟨H_set, hH, hA_coset⟩,
+    change H_set ∈ gal_basis K L at hH,
+    rw mem_gal_basis_iff at hH,
+    rcases hH with ⟨H, ⟨E, hE, hHE⟩, hHH_set⟩,
+    refine ⟨E, hE, _⟩,
+    rw [hHE, hHH_set],
+    exact hA_coset },
+  rcases hA_coset_cont_H with ⟨E, h_findim, hEA⟩,
+  have hEA' : left_coset σ E.fixing_subgroup ⊆ A,
+  { rintros x ⟨y, hy, hyx⟩,
+    change σ * y = x at hyx,
+    specialize hEA hy,
+    rcases hEA with ⟨a, ha, hay⟩,
+    change σ⁻¹ * a = y at hay,
+    rw inv_mul_eq_iff_eq_mul at hay,
+    rw [← hyx, ← hay],
+    exact ha },
```

Comment:
There's certainly more golfing to be done on this lemma. But here's a golf of the first half.
```suggestion
  change ↑f ≤ nhds σ,
  rw [←map_mul_left_nhds_one, group_filter_basis.nhds_one_eq],
  apply filter.le_map,
  suffices : ∀ s ∈ gal_basis K L, has_mul.mul σ '' s ∈ (f : filter (L ≃ₐ[K] L)),
  { rintros s ⟨t, ht, hs⟩,
    have key := this t ht,
    exact filter.mem_of_superset key (set.image_subset _ hs) },
  rintros - ⟨-, ⟨E, h_findim, rfl⟩, rfl⟩,
```